### DISCUSSION TITLE
GPL license: source code request

### DISCUSSION THREAD

**Discussion by matterhorner** [blinksh/blink#1776] [Support]
Blink was once an open source initiative and greatly benefited from utilizing the GPL-licensed Mosh, granting it a distinct advantage over other proprietary tools in the industry.

Despite efforts to persuade them, the Mosh developers refused to switch from the GPL to a more lenient license, as evidenced in this issue on GitHub: https://github.com/mobile-shell/mosh/issues/963.

However, it seems that the current version available on the AppStore is far more advanced than what is publicly available on GitHub, and might be in violation of the Mosh licensing terms: 

https://github.com/mobile-shell/mosh/blob/master/COPYING.iOS.

It’s unclear when Blink began to violate the GPL, but it’s evident that they continue to heavily publicize their usage of Mosh while keeping all recent advancements under lock and key.

A perusal of the project’s history indicates that the project went private with the 16.0 release, as announced here: 

https://github.com/blinksh/blink/discussions/1570. 

Nevertheless, Mosh remains an integral component of Blink and necessitates that the entirety of the application adhere to the GPLv3.

I hereby make an official appeal for the source code of the current iteration of BlinkShell that is accessible through the App Store, pursuant to Section 6 of the GPLv3.

**Comment by carloscabanero** [comment]
Hi! This was recently discussed in our subreddit. 

We released the code a while ago but with our latest changes became hard to compile by everyone, so we unrolled things to a point where they were easier. 

Want to clarify, as you are pointing that, that we have not violated the GPL. The code with latest changes has  always been available to anyone who requested it.

Anyway, we figured out a way to publish everything and just disable the parts that don’t make sense through some new flags and have done additional work to separate some of our clients (like Build) to its own library.

It will be pushed tomorrow.

**Comment by matterhorner** [comment]
Many thanks, your effort is appreciated by me.

Now, when you say “just disable the things that are not making sense”, do you mean you will use compile-time flags or are you keeping some code to yourself?

I am assuming that we will be receiving the new “Build” script since it is part of the overall GPL product that is presently up on the store.

**Reply by carloscabanero** [reply]
It is everything as it is on the AppStore. By Build I meant “Blink Build” client, being able to disable it, like we did with RevenueCat and other libraries.

Glad you appreciate the extra effort that went on this.

**Comment by carloscabanero** [comment]
Just a heads up that the latest version is now available at 16-cleanup, and everything as-is in the AppStore version. We will keep `raw` still as the recommended version to compile until next release. There are things related to secure keys that require to have your own domain for them to work, etc... so we will let the community figure out what the best way forward with those is.

Please note that the code is always and has always been available to Blink users upon request.

I created also Issue https://github.com/blinksh/blink/issues/1777 so if this were to happen in the future, Blink users would know which commit to request.

**Reply by matterhorner** [reply]
Thank you sir

Very appreciated
